src/refs.cpp

```
@@ -36,16 +89,31 @@ References references_from_stream(T& stream) {
 
     std::string line, seq, name;
     bool eof = false;
+    size_t line_num = 0;
     do {
+        line_num += 1;
         eof = !bool{getline(stream, line)};
         if (eof || (!line.empty() && line[0] == '>')) {
             if (seq.length() > 0) {
                 to_uppercase(seq);
                 sequences.push_back(seq);
                 names.push_back(name);
+                seen_names.insert(name);
             }
             if (!eof) {
-                name = line.substr(1, line.find(' ') - 1); // cut at first space
+                name = get_rname_prefix(line.begin() + 1, line.end());
```

I’m wondering whether just extracting the longest prefix that matches the regex is the best thing to do here. If you have a header such as `>my_impörtant_sequence`, this would silently truncate the name to `my_imp`. I think splitting at tab or space and then erroring out if the part before the whitespace doesn’t match the regex would be a little bit less surprising.

---

Sure! That would be reasonable also. It might also be good to still emit a warning or an error if the name you get does not satisfy the rules for SAM.